### PR TITLE
Empty `get addons` returns empty json/yaml array instead of empty string

### LINKED ISSUE #4910: Empty get addons as Json output should be empty array
I'm using `eksctl get addons -o json | jq ...`  in a script to know what addons are already installed and which ones I need to install.

But when no addon is installed, I just get this:
```
2022-03-07 16:07:33 [ℹ]  Kubernetes version "1.21" in use by cluster "my-cluster"
2022-03-07 16:07:33 [ℹ]  getting all addons
2022-03-07 16:07:33 [ℹ]  no addons found
```

No actual Json string is generated meaning Jq will have nothing to parse and I need to add cumbersome logic to handle this case.

Shouldn't ` eksctl get addons -o json` (and maybe other eksctl commands) have a consistent format, and return an empty Json array instead of an empty string when a Json output is requested ?
```
eksctl version: 0.84.0
kubectl version: v1.21.8
OS: linux
```

### PR DESCRIPTION
**Pull request by gdlx**
### Description

Returns empty json/yaml array/map instead of an empty strings when sending empty slice/array/map to Json/Yaml printers.

I had to call `make` to avoid `Marshall` to return `null`. I turned empty objects into string array/map as they're empty anyway.

Fixes #4910 

### Checklist
- [X] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [X] Manually tested
- [X] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Review by aclevername — APPROVED** [review]
lgtm!

File: pkg/ctl/get/addon.go [hunk 1]
```diff
@@ -92,8 +92,15 @@ func getAddon(cmd *cmdutils.Cmd, params *getCmdParams) error {
 
 	if len(summaries) == 0 {
 		logger.Info("no addons found")
-		return nil
+		if params.output == "json" || params.output == "yaml" {
```
[the remark this answers is not in the document]
Comment (gdlx) [reply pkg/ctl/get/addon.go]: Ok, I've moved the logic into the printers.
I didn't find a simple way not to repeat the code.
Also, I create a **string** empty slice, as we know it's empty anyway.
And I've also fixed maps, after checking they're facing the same bug.

[the remark this answers is not in the document]
Comment (gdlx) [reply pkg/ctl/get/addon.go]: I'll try to add some printer tests.

[the remark this answers is not in the document]
Comment (gdlx) [reply pkg/ctl/get/addon.go]: Added some tests and updated the PR description.

[the remark this answers is not in the document]
Comment (Skarlso) [reply pkg/ctl/get/addon.go]: @gdlx Thank you for doing this!

**Review by Skarlso — APPROVED** [review]
lgtm! nice job, well done for figuring out a nice way to do this. :)

**Review by nikimanoledaki — APPROVED** [review]

**Pull request merged**